src/game/option_passport.c

```diff
@@ -97,6 +99,14 @@ static void Option_PassportInitSaveRequester(int16_t page_num);
 static void Option_PassportLoadGame(void);
 static void Option_PassportSelectLevel(void);
 
+void Option_PassportInitSavegameStrings(void)
+{
+    m_SavegameStrings = Memory_Alloc(g_Config.maximum_save_slots * g_SavegameRequester.item_text_len);
+
+    // this is casting to a char** from a char*
+    g_SavegameRequester.item_texts = &m_SavegameStrings[0];
```

Comment:
This works fine, but the compiler does not like it. I looked at other places that `item_texts` is [initalised](https://github.com/rr-/Tomb1Main/blob/91de66e03613f28237653e45ec2c358b9a5b0770/src/game/option_passport.c#L233), but doing a similar thing here causes a segmentation fault.

Reply:
You're declaring it as a double pointer, so each member should be a pointer too, whereas it's zero-initialised. 
To circumvent this, either allocate each member separately, like this:
```c
m_SavegameStrings = Memory_Alloc(g_Config.maxium_save_slots * sizeof(char*));
for (int i = 0; i < g_Config.maxium_save_slots; i++) {
    m_SavegameStrings[i] = Memory_Alloc(g_SavegameRequester.item_text_len);
}
```
or declare these strings as `char*` then remove the cast.
Are you sure that using `item_text_len` here is OK?

Reply:
That makes sense, still slightly in the mindset that the language will do everything for me :p

And yes, `item_text_len` probably isn't okay on second look.